### PR TITLE
Pirate language

### PR DESCRIPTION
**Pull request by xmikey555**
:cl: 
add: Added pirate language, equip a pirate hat to speak it.
/:cl:

I haven't made a PR here since #23676 but this is pretty simple, it just adds "pirate language" (language only spoken/readable by fellow piratespeakers that looks like pirate slang to others), and makes the pirate hat (yes, the one obtainable from the autodrobe) grant/remove the language.

also i have no experience in dream maker so i'm at least 87% sure i flubbed something up

thanks to qustinnus for helping me out

### CODE REVIEW

File: code/modules/language/piratespeak.dm [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+	speech_verb = "says"
+	ask_verb = "asks"
+	exclaim_verb = "exclaims"
+	key = "p"
+	space_chance = 100
+	default_priority = 90
+	syllables = list(
+	"arr","ahoy","rum","aye","blimey","booty","bucko","grog","treasure",
+	"me","scallywag","landlubber","poopdeck","ye","avast",
+	"shiver me timbers","matey","treasure"
```
[the remark this answers is not in the document]
Comment (JJRcop) [reply code/modules/language/piratespeak.dm]: Emags, sugar, Alexa Walker, cultists, bread, starvation, guns, Cook, transparent, backpacks, beards, Cargo Technician, port emergency storage, crabs, important!

**Comment by XDTM** [comment]
It should probably work like the chef's text replacement instead of being a mishmash of pirate terms

File: code/modules/clothing/head/misc.dm [hunk 2]
```diff
@@ -129,6 +129,20 @@
...
+	var/datum/language/piratespeak = new
+
+/obj/item/clothing/head/pirate/equipped(mob/user, slot)
+	if(slot == SLOT_HEAD)
+		user.grant_language(/datum/language/piratespeak)
+		to_chat(user, "You suddenly know how to speak like a pirate!")
+
+/obj/item/clothing/head/pirate/dropped(mob/user)
+	var/mob/living/carbon/human/H = user
+	if(H.get_item_by_slot(SLOT_HEAD) == src)
```
Comment (Qustinnus) [inline code/modules/clothing/head/misc.dm]: im not sure if this works

File: code/modules/clothing/head/misc.dm [hunk 3]
```diff
@@ -129,6 +129,20 @@
 	item_state = "pirate"
 	dog_fashion = /datum/dog_fashion/head/pirate
 
+/obj/item/clothing/head/pirate
+	var/datum/language/piratespeak = new
```
[the remark this answers is not in the document]
Comment (xmikey555) [reply code/modules/clothing/head/misc.dm]: pushed

**Comment by Farquaar** [comment]
A couple more essential pirate words that were omitted:
yarr, land ho, yo ho ho, me hearties, aye, bilge-rat, Davy Jones' Locker, dead men tell no tales, heave ho, monkey, parrot, peg-leg, X marks the spot, 50 paces, dairy-farmer, island, Spanish-galleon, gold, dubloons
Pirate culture is beautiful and I want to see it implemented in its full majesty

**Pull request merged**